sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -773,39 +571,23 @@ def test_one_hot_encoder_drop_manual():
                        enc.inverse_transform(trans))
 
 
-def test_one_hot_encoder_invalid_params():
-    enc = OneHotEncoder(drop='second')
-    assert_raises_regex(
-        ValueError,
-        "Wrong input for parameter `drop`.",
-        enc.fit, [["Male"], ["Female"]])
-
-    enc = OneHotEncoder(handle_unknown='ignore', drop='first')
-    assert_raises_regex(
-        ValueError,
-        "`handle_unknown` must be 'error'",
-        enc.fit, [["Male"], ["Female"]])
-
-    enc = OneHotEncoder(drop='first')
-    assert_raises_regex(
-        ValueError,
-        "The handling of integer data will change in version",
-        enc.fit, [[1], [2]])
-
-    enc = OneHotEncoder(drop='first', categories='auto')
-    assert_no_warnings(enc.fit_transform, [[1], [2]])
-
-    enc = OneHotEncoder(drop=np.asarray('b', dtype=object))
-    assert_raises_regex(
-        ValueError,
-        "Wrong input for parameter `drop`.",
-        enc.fit, [['abc', 2, 55], ['def', 1, 55], ['def', 3, 59]])
-
-    enc = OneHotEncoder(drop=['ghi', 3, 59])
-    assert_raises_regex(
-        ValueError,
-        "The following categories were supposed",
-        enc.fit, [['abc', 2, 55], ['def', 1, 55], ['def', 3, 59]])
+@pytest.mark.parametrize(
+    "X_fit, params, err_msg",
+    [([["Male"], ["Female"]], {'drop': 'second'},
+     "Wrong input for parameter `drop`"),
+     ([["Male"], ["Female"]], {'drop': 'first', 'handle_unknown': 'ignore'},
+     "`handle_unknown` must be 'error'"),
+     ([['abc', 2, 55], ['def', 1, 55], ['def', 3, 59]],
+      {'drop': np.asarray('b', dtype=object)},
+     "Wrong input for parameter `drop`"),
+     ([['abc', 2, 55], ['def', 1, 55], ['def', 3, 59]],
+      {'drop': ['ghi', 3, 59]},
+     "The following categories were supposed")]
+)
```

Comment:
I personally don't find this more readable than the original ... (but that is subjective of course :-))

Reply:
I'll keep it for this time :)